components/proxy/src/main/java/com/hotels/styx/proxy/plugin/InstrumentedPlugin.java

```diff
@@ -42,21 +43,23 @@
     private static final Logger LOGGER = getLogger(InstrumentedPlugin.class);
 
     private final NamedPlugin plugin;
-    private final SimpleCache<HttpResponseStatus, Meter> errorStatusMetrics;
-    private final SimpleCache<Class<? extends Throwable>, Meter> exceptionMetrics;
-    private final Meter errors;
+    private final SimpleCache<HttpResponseStatus, Counter> errorStatusMetrics;
+    private final SimpleCache<Class<? extends Throwable>, Counter> exceptionMetrics;
+    private final Counter errors;
 
     public InstrumentedPlugin(NamedPlugin plugin, Environment environment) {
         this.plugin = requireNonNull(plugin);
         requireNonNull(environment);
 
+        Metrics.addRegistry(environment.meterRegistry());
```

Comment:
You shouldn't need to add the environment registry to the global registry. This is not so much a problem for test code, but for application code you should just work with the registry that you are given. Then to create metrics you can use `Counter.builder(...)....register(registry)`